packages/types/src/index.ts

```diff
@@ -142,6 +142,16 @@ export enum SignatureType {
     NSignatureTypes,
 }
 
+/**
+ * The Signer Provider Type. Some Signer implementations use different message prefixes or implement different
+ * eth_sign behaviour. Note EthSign is compatible with the Ledger device.
+ */
+export enum SignerProviderType {
+    EthSign = 'ETH_SIGN',
+    Metamask = 'METAMASK',
+    Trezor = 'TREZOR',
```

Comment:
I think we should still include `Ledger` despite it being identical under-the-hood. Might give developers more certainty it'll work as expected when using Ledger?

Also, let's rename `EthSign` to `Other`, since it's not a signer name, and there are many more Signers out there but we should assume they work to-spec until someone raises an issue otherwise.

Reply:
Yeah I had Ledger here as well originally to signal the developer.  I'm not a huge fan of other since it is not that intuitive. EthSign is aligned with our SignatureTypes in contracts. Once Metamask fixes their bug the only availble options would be `Trezor` and `Other`